models/MARBL_joint_estimation/model_mod.f90

```diff
@@ -0,0 +1,654 @@
+integer :: state_qty_list(modelvar_table_height), &
+           param_qty_list(modelparams_table_height)
+real(r8):: state_clamp_vals(modelvar_table_height, 2), &
+           param_clamp_vals(modelparams_table_height, 2)
+logical :: update_var_list(modelvar_table_height), &
+           update_param_list(modelparams_table_height)
+
+module_initialized = .true.
+
+! Print module information to log file and stdout.
+call register_module(source)
+
+! Read values from the namelist
+
+call find_namelist_in_file("input.nml", "model_nml", iunit)
+read(iunit, nml = model_nml, iostat = io)
+
+call check_namelist_read(iunit, io, "model_nml")
+
+! Record the namelist values used for the run 
+if (do_nml_file()) write(nmlfileunit, nml=model_nml)
+if (do_nml_term()) write(     *     , nml=model_nml)
+
+! This time is both the minimum time you can ask the model to advance
+! (for models that can be advanced by filter) and it sets the assimilation
+! window.  All observations within +/- 1/2 this interval from the current
+! model time will be assimilated. If this is not settable at runtime 
+! feel free to hardcode it and remove from the namelist.
+assimilation_time_step = set_time(time_step_seconds, &
+                                  time_step_days)
+
+! setting up the DART state vector
+call verify_state_variables(model_state_variables, nfields, variable_table, &
+                            state_qty_list, state_clamp_vals, update_var_list)
+
+    state_dom_id = add_domain(template_file(1), nfields, &
+                              var_names = variable_table(1:nfields, 1), &
+                              kind_list = state_qty_list(1:nfields), &
+                              clamp_vals = state_clamp_vals, &
+                              update_list = update_var_list(1:nfields))
+
+! setting up the DART parameter vector
+if(estimate_params) then
+    call verify_state_variables(model_parameters, nfields, param_table, &
+                                param_qty_list, param_clamp_vals, update_param_list)
+
+    param_dom_id = add_domain(template_file(2), nfields, &
+                              var_names = param_table(1:nfields, 1), &
+                              kind_list = param_qty_list(1:nfields), &
+                              clamp_vals = param_clamp_vals, &
+                              update_list = update_param_list(1:nfields))
+end if
+
+call read_num_layers     ! setting the value of nz
+call read_ocean_geometry ! determining the basin depth
+
+end subroutine static_init_model
+
+!------------------------------------------------------------------
+! Reads the simulation length from a netCDF file.
+
+function read_model_time(filename)
+
+character(len=*), intent(in) :: filename
+type(time_type) :: read_model_time
+
+integer :: ncid
+character(len=*), parameter :: routine = 'read_model_time'
+real(r8) :: days
+type(time_type) :: mom6_time
+integer :: mom_base_date_in_days, mom_days
+
+mom_base_date_in_days = 139157 ! 1982 1 1 0 0
```

Comment:
Is this always Jan 1st 1982?

Reply:
Unless changes have been made to the MARBL date/time format, yes, it's always Jan 1st 1982.